pipeline/backend/kubernetes/pod.go

```diff
@@ -108,7 +108,11 @@ func podLabels(step *types.Step, config *config, options BackendOptions) (map[st
 		// Only copy user labels if allowed by agent config.
 		// Internal labels are filtered on the server-side.
 		if config.PodLabelsAllowFromStep || strings.HasPrefix(k, pipeline.InternalLabelPrefix) {
-			labels[k] = v
+			sanitizedValue, err := toDNSName(v)
+			if err != nil {
+				return nil, err
+			}
+			labels[k] = sanitizedValue
```

Comment:
```suggestion
			labels[k], err = toDNSName(v)
			if err != nil {
				return labels, err
			}
```

Can we not skip the extra spep here as well? I would also return `labels` on errors instead of `nil` to follow the existing implementation.

Reply:
fixed in 41dd58c573284b7c7e2c43b71e45403e0892035a

Reply:
Only partially :) The suggestion also drops the extra assignment `sanitizedValue, err := toDNSName(v)`